src/isomorphic/devtools/ReactComponentTreeDevtool.js

```diff
@@ -78,10 +79,11 @@ var ReactComponentTreeDevtool = {
           'Expected onMountComponent() to fire for the child ' +
           'before its parent includes it in onSetChildren().'
         );
-
-        if (prevChildIDs.indexOf(nextChildID) === -1) {
-          nextChild.parentID = id;
-        }
+        nextChild.parentID = id;
+        // TODO: invariant(nextChild.parentID === id) makes sense but doesn't
```

Comment:
Can we add an invariant for the cases where we their `parentId` is not missing? It being missing is just a subset of cases, is it not? I mean something like `invariant(nextChild.parentID == null || nextChild.parentID === id)`